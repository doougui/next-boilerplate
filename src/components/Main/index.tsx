import * as S from './styles';

function Main({
  title = 'Next.js Boilerplate',
  description = 'Initial boilerplate for Next.js with TypeScript projects',
}) {
  return (
    <S.Wrapper>
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
      <S.Illustration
        src="/img/hero-illustration.svg"
        alt="Um desenvolvedor de frente para uma tela com cÃ³digo."
      />
    </S.Wrapper>
  );
}

export default Main;
