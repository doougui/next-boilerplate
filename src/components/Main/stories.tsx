import Main from '.'
import { Story, Meta } from '@storybook/react/types-6-0'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'default title',
    description: 'default description'
  }
} as Meta

export const Basic: Story = (args) => <Main {...args} />

Basic.args = {
  title: 'React Avançadooooo',
  description: 'Descrição foda'
}

export const Default: Story = (args) => <Main {...args} />
