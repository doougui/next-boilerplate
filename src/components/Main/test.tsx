import { render, screen } from 'utils/tests';

import Main from '.';

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', { name: /next\.js boilerplate/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render colors correctly', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({ 'background-color': '#111111' });
  });
});
