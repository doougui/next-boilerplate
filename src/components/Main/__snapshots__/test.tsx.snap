// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="styles__Wrapper-sc-1fpvr54-0 dkyCmT"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado"
    class="styles__Logo-sc-1fpvr54-1 iNkjXk"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-sc-1fpvr54-2 dDTJbC"
  >
    React Avançado
  </h1>
  <h2
    class="styles__Description-sc-1fpvr54-3 kBlFtR"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela com código."
    class="styles__Illustration-sc-1fpvr54-4 dHRuKw"
    src="/img/hero-illustration.svg"
  />
</main>
`
